spec/controllers/api/public/v1/providers/courses_controller_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe API::Public::V1::Providers::CoursesController do
+  let(:provider) { create(:provider) }
+  let(:recruitment_cycle) { provider.recruitment_cycle }
+
+  describe "#index" do
+    context "when there are no courses" do
+      it "returns empty array of data" do
+        get :index, params: {
```

Comment:
can we put the `get` in a before block that way we setup the tests in a an arrange/act/assert structure. that way the it blocks are just asserting on what has been arranged/performed?

Reply:
cool, i've changed all these